src/test/suite/compile.test.ts

```diff
@@ -0,0 +1,106 @@
+import * as assert from 'assert';
+import { after, before, describe, it, beforeEach } from 'mocha';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+
+import { Uri, window, commands, workspace, TextDocument, TextEditor, Position, SnippetString } from 'vscode';
+
+const logPath: string = '/tmp/vscode-inmanta.log';
+const workspaceUri: Uri = Uri.file(path.resolve(__dirname, '../../../src/test/workspace'));
+const libsPath: string = path.resolve(workspaceUri.fsPath, 'libs');
+const envPath: string = path.resolve(workspaceUri.fsPath, '.env');
+
+const modelUri: Uri = Uri.file(path.resolve(workspaceUri.fsPath, 'main.cf'));
+
+function waitForCompile(timeout: number): Promise<boolean> {
+	const start = Date.now();
+	return new Promise<boolean>((resolve, reject) => {
+		const readLogInterval = setInterval(() => {
+			if (Date.now() - start > timeout) {
+				reject(new Error("Timeout reached"));
+			} else {
+				fs.ensureFileSync(logPath);
+				fs.readFile(logPath, 'utf-8', (err, data) => {
+					if (err) {
+						console.log(err);
+					} else if (data.includes('Compilation succeeded')) {
+						clearInterval(readLogInterval);
+						resolve(true);
+					} else if (data.includes('Compilation failed')) {
+						clearInterval(readLogInterval);
+						resolve(false);
+					}
+				});
+			}
+		}, 500);
+	});
+}
+
+describe('Compile checks', () => {
+	const tests = [
+		{ source: 'valid.cf', succeed: true },
+		{ source: 'invalid.cf', succeed: false }
+	];
+
+	before((done) => {
+		commands.executeCommand('vscode.openFolder', workspaceUri).then(done);
+	});
+
+	beforeEach((done) => {
+		Promise.all([
+			fs.writeFile(logPath, ""),
+			fs.remove(libsPath),
+			fs.remove(envPath),
+			fs.remove(modelUri.fsPath),
+		]).then(async values => {
+			await commands.executeCommand('workbench.action.closeActiveEditor');
+			done();
```

Comment:
If you chain a `then()` to the `executeCommand` and call the `done` callback there, can't the explicit waiting in the test case be removed?

Reply:
No because the `executeCommand` actually gives an argument, and done will refuse any call with an argument.